### PR TITLE
Fixes 1101

### PR DESCRIPTION
**Pull request by ZheHuang96**
Hi Rikard,

This PR is to fix issue#1101.

Best regards,
Zhe Huang

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,3 +1,11 @@
+Next release
+-------------------
+
+Bugfixes
+========
+
+* bd3d868f Set dofv to nan when candidate_model is None (#1101)
```
Comment (pharmpy-dev-123) [inline CHANGELOG.rst:7]: :warning: Given this is a PR this commit handle might not be the final one.

File: src/pharmpy/modeling/summarize_individuals.py [hunk 2]
```diff
@@ -132,8 +132,14 @@ def ofv(model: Model) -> Union[Series, float]:
...
-def dofv(parent_model: Union[Model, None], candidate_model: Model) -> Union[Series, float]:
-    return np.nan if parent_model is None else ofv(parent_model) - ofv(candidate_model)
+def dofv(
+    parent_model: Union[Model, None], candidate_model: Union[Model, None]
+) -> Union[Series, float]:
+    return (
+        np.nan
+        if parent_model is None or candidate_model is None
+        else ofv(parent_model) - ofv(candidate_model)
+    )
```
Comment (pharmpy-dev-123) [inline src/pharmpy/modeling/summarize_individuals.py:142]: LGTM!

File: CHANGELOG.rst (hunk 1 above)
Comment (ZheHuang96) [reply CHANGELOG.rst:7]: Yes, I see.  I am thinking is there a better way to handle it so that we will not forget to add each fix in the CHAGNELOG? 

> ⚠️ Given this is a PR this commit handle might not be the final one.

Comment (pharmpy-dev-123) [reply CHANGELOG.rst:7]: Put ??? instead?

File: src/pharmpy/modeling/summarize_individuals.py (hunk 2 above)
Comment (pharmpy-dev-123) [reply src/pharmpy/modeling/summarize_individuals.py:142]: Actually, if this is meant to fix #1101 I think the problem comes from `model.modelfit_results.individual_ofv` being `None` instead of `np.nan`.

**Review by pharmpy-dev-123 — CHANGES_REQUESTED** [review]

**Comment by ZheHuang96** [comment]
Hi @pharmpy-dev-123 ,

According to the discussion, I set up a test to reproduce the error message. But it is not a 'real' test of the functions `dofv()` and `ofv()` , so I am not sure this test should be kept or removed.

**Comment by pharmpy-dev-123** [comment]
> Hi @pharmpy-dev-123 ,
> 
> According to the discussion, I set up a test to reproduce the error message. But it is not a 'real' test of the functions `dofv()` and `ofv()` , so I am not sure this test should be kept or removed.

I think the test should be changed to test the `ofv` function rather than the `dofv` function.

File: CHANGELOG.rst (hunk 1 above)
Comment (rikardn) [reply CHANGELOG.rst]: It is enough to have the issue number. If the issue #1101 is mentioned here it will have a connection.

**Comment by ZheHuang96** [comment]
> > Hi @pharmpy-dev-123 ,
> > According to the discussion, I set up a test to reproduce the error message. But it is not a 'real' test of the functions `dofv()` and `ofv()` , so I am not sure this test should be kept or removed.
> 
> I think the test should be changed to test the `ofv` function rather than the `dofv` function.

So you want the test for this line:  `np.nan if res is None or res.individual_ofv is None else res.individual_ofv` ? I am not sure whether I can create a dummy model or not, because I think if we want to test the whole functionality, we should have a  model which individual ofv is None.

File: tests/modeling/test_summarize_individuals.py [hunk 3]
```diff
@@ -42,3 +44,10 @@ def test_tflite_not_installed(pheno_path, monkeypatch):
         monkeypatch.setitem(sys.modules, 'tflite_runtime', None)
         df = summarize_individuals([model])
         assert df['predicted_dofv'].isnull().all().all()
+
+
+def test_ofv(pheno_path):
+    parent_model = read_model(pheno_path)
+    candidate_model = Model()
```
Comment (pharmpy-dev-123) [inline tests/modeling/test_summarize_individuals.py:51]: Can you also test with a model that has an empty `modelfit_results` object instead of `None`?

**Review by pharmpy-dev-123 — CHANGES_REQUESTED** [review]

File: tests/modeling/test_summarize_individuals.py (hunk 3 above)
Comment (ZheHuang96) [reply tests/modeling/test_summarize_individuals.py:51]: So you want a model which is never fitted?

Comment (ZheHuang96) [reply tests/modeling/test_summarize_individuals.py]: Now we have a none modelfit_results:)

**Review by pharmpy-dev-123 — APPROVED** [review]
Once unit tests are fixed by @rikardn I think these new tests will pass and it is good to merge.

**Comment by ZheHuang96** [comment]
I will rebase it now! Many thanks!
